win32ss/gdi/ntgdi/dibobj.c

```diff
@@ -129,34 +127,32 @@ CreateDIBPalette(
         {
             /* The colors are an array of RGBQUAD values */
             RGBQUAD *prgb = (RGBQUAD*)((PCHAR)pbmi + pbmi->bmiHeader.biSize);
+            RGBQUAD colors[256] = {0};
 
             // FIXME: do we need to handle PALETTEINDEX / PALETTERGB macro?
 
-            /* Loop all color indices in the DIB */
-            for (i = 0; i < cColors; i++)
+            /* Use SEH to verify we can READ prgb[] succesfully */
+            _SEH2_TRY
             {
-                /* User SEH to verify READ success */
-                Status = STATUS_SUCCESS;
-                _SEH2_TRY
-                {
-                    rgb = prgb[i];
-                }
-                _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
+                for(i = 0; i < cColors; ++i)
```

Comment:
is `cColors` guaranteed to be less than 256 ? 
Also, `RtlCopyMemory(colors, prgb, cColors * sizeof(colors[0]))` would look better to me.

Reply:
cColors will be less than or equal to 256. Looking at the code above in Line #60 we have this:
```
    if (cBitsPixel <= 8)
```
Which is a prerequisite of reaching our code at all. So the maximum value of cBitsPixel is 8.
Next in Line #63, we have the following:
```
        cColors = 1 << cBitsPixel;
```
So taking the maximum value of '8' for cBitsPixel and shifting 1 left this '8' times gives us 256.
This guarantees that the cColors value will never be greater than 256.

I will change from using the for look to your RtlCopyMemory since it looks better to you.

Reply:
> Now in Line#78 we have this:
>            cColors = min(cColors, pbmi->bmiHeader.biClrUsed);

This by itself doesn't guarantee anything, as biClrUsed can be anything (it's a DWORD value).

Reply:
@HBelusca, a 'min' function must take the lowest value, so the maximum of any one value must be the maximum for the 'min' function. So if we have min(256, AnythingElse), we know that this cannot return anything greater than 256. Correct?

Reply:
Ah yes, indeed, if either one of the "parameters" of the min macro is already ensured to be not greater than 256, then it should return a value <= 256.

Reply:
Thanks for noting your agreement with me on this. :)